Detectors/TRD/workflow/src/TRDGlobalTrackingSpec.cxx

```diff
@@ -0,0 +1,201 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// @file   TRDGlobalTrackingSpec.cxx
+
+#include "TRDWorkflow/TRDGlobalTrackingSpec.h"
+
+#include "TRDBase/Geometry.h"
+
+#include "DetectorsCommonDataFormats/NameConf.h"
+#include "CommonConstants/LHCConstants.h"
+#include "DetectorsBase/GeometryManager.h"
+#include "DetectorsBase/Propagator.h"
+#include "ReconstructionDataFormats/TrackTPCITS.h"
+#include "DataFormatsTRD/Tracklet64.h"
+#include "DataFormatsTRD/TriggerRecord.h"
+
+// GPU header
+#include "GPUReconstruction.h"
+#include "GPUChainTracking.h"
+#include "GPUSettings.h"
+#include "GPUDataTypes.h"
+#include "GPUTRDDef.h"
+#include "GPUTRDTrack.h"
+#include "GPUTRDTrackletWord.h"
+#include "GPUTRDInterfaces.h"
+#include "GPUTRDGeometry.h"
+
+using namespace o2::framework;
+using namespace o2::gpu;
+
+namespace o2
+{
+namespace trd
+{
+
+void TRDGlobalTracking::init(InitContext& ic)
+{
+  //-------- init geometry and field --------//
+  o2::base::GeometryManager::loadGeometry();
+  o2::base::Propagator::initFieldFromGRP(o2::base::NameConf::getGRPFileName());
+  auto geo = Geometry::instance();
+  geo->createPadPlaneArray();
+  geo->createClusterMatrixArray();
+  const GeometryFlat geoFlat(*geo);
+
+  //-------- init GPU reconstruction --------//
+  GPUSettingsEvent cfgEvent;                 // defaults should be ok
+  GPUSettingsRec cfgRec;                     // don't care for now, NWaysOuter is set in here for instance
+  GPUSettingsProcessing cfgDeviceProcessing; // also keep defaults here, or adjust debug level
+  cfgDeviceProcessing.debugLevel = 5;
+  GPURecoStepConfiguration cfgRecoStep;
+  cfgRecoStep.steps = GPUDataTypes::RecoStep::NoRecoStep;
+  cfgRecoStep.inputs.clear();
+  cfgRecoStep.outputs.clear();
+  mRec = GPUReconstruction::CreateInstance("CPU", true);
+  mRec->SetSettings(&cfgEvent, &cfgRec, &cfgDeviceProcessing, &cfgRecoStep);
+
+  mChainTracking = mRec->AddChain<GPUChainTracking>();
+
+  mTracker = new GPUTRDTracker();
+  mTracker->SetNCandidates(1); // must be set before initialization
+  mTracker->SetProcessPerTimeFrame();
+  mTracker->SetNMaxCollisions(1000); // max number of collisions within a time frame which can be processed
+
+  mRec->RegisterGPUProcessor(mTracker, false);
+  mChainTracking->SetTRDGeometry(&geoFlat);
+  if (mRec->Init()) {
+    LOG(FATAL) << "GPUReconstruction could not be initialized";
+  }
+
+  // configure the tracker
+  //mTracker->EnableDebugOutput();
+  //mTracker->StartDebugging();
+  mTracker->SetPtThreshold(0.5);
```

Comment:
I think in the long run, we should move these parameters that will be relevant to `GPUSettingsRec` and then use `--configKeyValues` to set them, as it is already done in the TPC workflow. But I would do this in an independent PR, and first try to get the workflow going.